image/image.go

```diff
@@ -0,0 +1,255 @@
+type ErrInvalidImageAPIResponse struct {
+	actualCode int
+	uri        string
+	body       string
+}
+
+// Error should be called by the user to print out the stringified version of the error
+func (e ErrInvalidImageAPIResponse) Error() string {
+	return fmt.Sprintf("invalid response: %d from image api: %s, body: %s",
+		e.actualCode,
+		e.uri,
+		e.body,
+	)
+}
+
+// Code returns the status code received from image api if an error is returned
+func (e ErrInvalidImageAPIResponse) Code() int {
+	return e.actualCode
+}
+
+// compile time check that ErrInvalidImageAPIResponse satisfies the error interface
+var _ error = ErrInvalidImageAPIResponse{}
+
+// Client is a image api client which can be used to make requests to the server
+type Client struct {
+	cli dphttp.Clienter
+	url string
+}
+
+// closeResponseBody closes the response body and logs an error containing the context if unsuccessful
+func closeResponseBody(ctx context.Context, resp *http.Response) {
+	if err := resp.Body.Close(); err != nil {
+		log.Event(ctx, "error closing http response body", log.ERROR, log.Error(err))
+	}
+}
+
+// NewAPIClient creates a new instance of Client with a given image api url and the relevant tokens
+func NewAPIClient(imageAPIURL string) *Client {
+	hcClient := healthcheck.NewClient(service, imageAPIURL)
+
+	return &Client{
+		cli: hcClient.Client,
+		url: imageAPIURL,
+	}
+}
+
+// NewAPIClientWithMaxRetries creates a new instance of Client with a given image api url and the relevant tokens,
+// setting a number of max retires for the HTTP client
+func NewAPIClientWithMaxRetries(imageAPIURL string, maxRetries int) *Client {
+	hcClient := healthcheck.NewClient(service, imageAPIURL)
+	if maxRetries > 0 {
+		hcClient.Client.SetMaxRetries(maxRetries)
+	}
+
+	return &Client{
+		cli: hcClient.Client,
+		url: imageAPIURL,
+	}
+}
+
+// Checker calls image api health endpoint and returns a check object to the caller.
+func (c *Client) Checker(ctx context.Context, check *health.CheckState) error {
+	hcClient := healthcheck.Client{
+		Client: c.cli,
+		URL:    c.url,
+		Name:   service,
+	}
+
+	return hcClient.Checker(ctx, check)
+}
+
+// GetImages returns the list of images
+func (c *Client) GetImages(ctx context.Context) (m Images, err error) {
+	uri := fmt.Sprintf("%s/images", c.url)
+
+	clientlog.Do(ctx, "retrieving images", service, uri)
+
+	req, err := http.NewRequest(http.MethodGet, uri, nil)
+	if err != nil {
+		return
+	}
+
+	resp, err := c.cli.Do(ctx, req)
+	if err != nil {
+		return
+	}
+	defer closeResponseBody(ctx, resp)
+
+	if resp.StatusCode != http.StatusOK {
+		err = NewImageAPIResponse(resp, uri)
+		return
+	}
+
+	b, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return
+	}
+
+	if err = json.Unmarshal(b, &m); err != nil {
+		return
+	}
+
+	return
+}
+
+// PostImage performs a 'POST /images' with the provided NewImage
+func (c *Client) PostImage(ctx context.Context, data NewImage) (m Image, err error) {
+	payload, err := json.Marshal(data)
```

Comment:
If we decide to use the iota enum, here we could verify that the provided Image state is a valid state